### PR TITLE
Feat/Apply Online button description

### PR DESCRIPTION
**Pull request by msrezaie**
What (if anything) did you refactor?
- Made the 'Apply Online' button get its text from the newly added `apply_button_description` field of a program.

'Apply Online' button previously:
image

Now with customized button description:
Screenshot 2024-11-08 152407

### CODE REVIEW

**Review by CalebPena — CHANGES_REQUESTED** [review]
We want the default to still be "Apply Online". Can you check if there is an `apply_button_description`, and make the button say "Apply Online" if the `default_message` is an empty string?

**Comment by msrezaie** [comment]
@CalebPena the `apply_button_description` is the newly added field ([PR](https://github.com/Gary-Community-Ventures/benefits-api/pull/630)) into the Programs table to allow a program instance to have a unique description for the apply button. In the PR for this, the translation label getting attached to the `apply_button_description` field will be populated with 'Apply Online' by default: [reference](https://github.com/Gary-Community-Ventures/benefits-api/pull/630/files#diff-8e858524b8253844bdcd2c8216f67bd80182c79daf50b16caecfcd8b50936fbcR12)

**Comment by CalebPena** [comment]
@msrezaie The default in [the backend pr](https://github.com/Gary-Community-Ventures/benefits-api/pull/630/files#diff-8e858524b8253844bdcd2c8216f67bd80182c79daf50b16caecfcd8b50936fbcR12) does not translate the text. Additionally, I think that we want to simplify the admin experience by always defaulting to "Apply Online", and then the admin only has to override it for the programs where it is different.

**Comment by msrezaie** [comment]
@CalebPena I added bulk translate logic in the migration ([reference](https://github.com/Gary-Community-Ventures/benefits-api/pull/630/commits/f2bc6ae5f1ac7cb1270ad062bc42850a8b401311)) so now all of the `apply_button_description` text will also be translated. I've also added a conditional in [here](https://github.com/Gary-Community-Ventures/benefits-calculator/pull/1363/commits/8734b27d084989b9e17e531e0ba13ff92ecdaae5#diff-53ab548d6e85f98f945320b622e4f30f0fe966fabecc4176b704051dedc2c505R170) to check if `apply_button_description` label is an empty string then use the previous label for the button. The only problem with this is that whenever a new program is added manually all of its fields will default to a `[PLACEHOLDER]` text. Any recommendations for a better fix for this?

File: src/Components/Results/ProgramPage/ProgramPage.tsx [hunk 1]
```diff
@@ -166,7 +166,11 @@ const ProgramPage = ({ program }: ProgramPageProps) => {
       </div>
       <div className="apply-button-container">
         <a className="apply-online-button" href={program.apply_button_link.default_message} target="_blank">
-          <FormattedMessage id="results.apply-online" defaultMessage="Apply Online" />
+          {program.apply_button_description.default_message == "" ? (
+            <FormattedMessage id="results.apply-online" defaultMessage="Apply Online" />
+          ) : (
+            <FormattedMessage id={program.apply_button_description.label} defaultMessage="Apply Online" />
```
Comment (CalebPena) [inline src/Components/Results/ProgramPage/ProgramPage.tsx:172]:
```suggestion
            <ResultsTranslate translation={program.apply_button_description} />
```

**Review by CalebPena — APPROVED** [review]
LGTM. There is one small improvement that you can make if you want before merging.

File: src/Components/Results/ProgramPage/ProgramPage.tsx (hunk 1 above)
Comment (CalebPena) [reply src/Components/Results/ProgramPage/ProgramPage.tsx:172]: Actually, I am going to commit this myself so we can release this today.

Comment (msrezaie) [reply src/Components/Results/ProgramPage/ProgramPage.tsx]: Thanks for that.

**Pull request merged**